indra/llcommon/llapr.cpp

```diff
@@ -574,6 +575,9 @@ S32 LLAPRFile::readEx(const std::string& filename, void *buf, S32 offset, S32 nb
 S32 LLAPRFile::writeEx(const std::string& filename, const void *buf, S32 offset, S32 nbytes, LLVolatileAPRPool* pool)
 {
     LL_PROFILE_ZONE_SCOPED;
+    if (LLApp::isFileReadOnlyMode())
```

Comment:
Operations will just fail with no reason yet resources for writing will keep allocating? Also this isn't the only way of writing to cache, textures use Apr because it's supposed to be thread friendly.

Reply:
What do you mean writing "read only"? The firld LLAppViewer::mSecondInstance ?

mSecondInstance is not used exclusively to prevent writing to the disk,
it seems to be used in absolutely different algorithms

And writing to the disk could happen in many cases without testing this flag

Also I'm not sure that we may turn it on in the middle of the viewer session
Viewer with mSecondInstance uses other directories for writing files

That's why I decided that we'd better introduce a new separate flag,
semantically corresponding to the general purpose: prevent storing data to the disk

And yes, there are many points where viewer writes files,
so the current commit is just a declaration of the idea, but not a final it's implementation

It  is still a draft